gammapy/time/lightcurve.py

```diff
@@ -314,11 +321,204 @@ def create_fixed_time_bin(time_step, spectrum_extraction):
+                 obs_properties['A_off'][n + tmp]
+        time += (1 - obs_properties['deadtime'][n + tmp]) * (float(time_holder[i][0]) - float(time_holder[xm1][0]))
+        alpha = time / alpha
+        return alpha
+
+    def make_time_intervals_min_significance(self, significance, significance_method, energy_range,
+                                             spectrum_extraction, separators=None):
+        """
+
+        Create time intervals such that each bin of a light curve reach a given significance
+
+        The function create a list of time associated with identifiers called time_holder.
+        The identifiers can be 'on' for the on events, 'off' for the off events, 'start' for the start of an
+        observation, 'end for the end of an observation and 'break' for a separator.
+
+        Parameters
+        ----------
+        significance : float
+            Target significance for each light curve point
+        significance_method : {'lima', 'simple'}
+            Significance method (see `~gammapy.stats.significance_on_off`)
+        energy_range : `~astropy.units.Quantity`
+            True energy range to evaluate integrated flux (true energy)
+        spectrum_extraction : `~gammapy.spectrum.SpectrumExtraction`
+            Contains statistics, IRF and event lists
+        separators : `list` of float
+            Contains a list of time (in MJD) to break the current point creation
+
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            Table of time intervals  and information about their content : on/off events, alpha, significance
+
+        Examples
+        --------
+        extract intervals for light curve :
+            intervals = list(zip(table['t_start'], table['t_stop']))
+
+        """
+
+        time_holder = []
+        obs_properties = []
+        n_obs = 0
+
+        if separators is not None:
+            for time in separators:
+                time_holder.append([Time(time, format='mjd').tt.mjd, 'break'])
+
+        for obs in spectrum_extraction.obs_list:
+            # start and end will need to be redefined once the data storage format is fixed
+            time_holder.append([obs.events.observation_time_start.tt.mjd, 'start'])
+            time_holder.append([obs.events.observation_time_end.tt.mjd, 'end'])
+            obs_properties.append(dict(deadtime=obs.observation_dead_time_fraction,
+                                       A_off=spectrum_extraction.bkg_estimate[n_obs].a_off))
+            n_obs += 1
+        obs_properties = Table(rows=obs_properties)
+
+        # prepare the on and off photon list as in the flux point computation -> should be updated accordingly
+        for t_index, obs in enumerate(self.obs_list):
+            on, off = self._create_and_filter_onofflists(t_index=t_index, energy_range=energy_range)
+
+            for time in on.time:
+                time_holder.append([time.tt.mjd, 'on'])
```

Comment:
I think here you're executing `time.tt.mjd` for every single event?
This is pretty inefficient, especially if the `time` timescale is not `tt` and a time scale transformation has to be computed.
To avoid it you can do `time = time.tt.mjd` at the array level once at the start, no?

Reply:
I tried at array level but it didn't work.
Isn't .tt.mjd equivalent to .value but with the added correction in case of different scales?

Reply:
I think `.value` can give floats, or strings, or datetime objects, depending on the `format` of the time object.
```
In [3]: Time.now()
Out[3]: <Time object: scale='utc' format='datetime' value=2018-05-02 10:39:04.151454>

In [4]: Time.now().value
Out[4]: datetime.datetime(2018, 5, 2, 10, 39, 10, 801319)
```
So I don't think just calling `time.value` is a good idea.

I think either `time.tt.mjd` for all times (from the events and lightcurve, and user inputs) is OK, or probably an even better general strategy for Gammapy would be to go to "MET", i.e. mission elapsed time in seconds when working with times as floats, i.e. do `met = (time - events.time_ref).sec` and work with those floats. This is what I would do when working with event times; it's just that for lightcurves I think people often use MJD, i.e. days with a different fixed reference time that doesn't depend on the event list. So I'm not sure here.

But anyways, my comment was to avoid this call for each event separately, but do it for the array of events at the top. That should work just fine (you just say "didn't work") either way and be more effiicent and simpler that way, because it separates the time object (scale and format) handling from the algorithm, but doing a sandwich (Time -> float at start, then the algorithm, then float -> Time at the end). It's not super important, but I do think it would be overall cleaner to write it that way.

Reply:
I will try again, but if I can't make it work  I will keep it as it is.